packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -249,45 +259,36 @@ export class SCWSigner implements Signer {
 
         const account = response.accounts.at(0);
         const capabilities = account?.capabilities;
-        if (capabilities?.addSubAccount || capabilities?.getSubAccounts) {
-          const capabilityResponse =
-            capabilities?.addSubAccount ?? capabilities?.getSubAccounts?.[0];
-          assertSubAccount(capabilityResponse);
+        const requestCapabilities = (request as WalletConnectRequest).params?.[0]?.capabilities;
+
+        if (capabilities?.subAccounts) {
+          const capabilityResponse = capabilities?.subAccounts;
+          assertArrayPresence(capabilityResponse, 'subAccounts');
+          assertSubAccount(capabilityResponse[0]);
           store.subAccounts.set({
-            address: capabilityResponse?.address,
-            factory: capabilityResponse?.factory,
-            factoryData: capabilityResponse?.factoryData,
+            address: capabilityResponse[0].address,
+            factory: capabilityResponse[0].factory,
+            factoryData: capabilityResponse[0].factoryData,
           });
         }
         let accounts_ = [this.accounts[0]];
 
         const subAccount = store.subAccounts.get();
         const subAccountsConfig = store.subAccountsConfig.get();
-        if (subAccount?.address) {
-          // Sub account is always at index 0
-          accounts_ = [subAccount.address, ...this.accounts];
 
-          // Also update the accounts store if automatic sub accounts are enabled
-          if (subAccountsConfig?.enableAutoSubAccounts) {
-            store.account.set({
-              accounts: accounts_,
-            });
-            this.accounts = accounts_;
-          }
-        }
+        // Sub account should be returned as a top level account if auto sub accounts are enabled
+        const shouldUseSubAccount =
+          subAccountsConfig?.enableAutoSubAccounts || !!requestCapabilities?.addSubAccount;
```

Comment:
should be `requestCapabilities.subAccounts` here?

Reply:
no we still have the `addSubAccount` capability